src/components/AddItems.js

```diff
@@ -9,9 +9,13 @@ const arrayUnion = firebase.firestore.FieldValue.arrayUnion;
 const AddItemsToList = () => {
   const [shoppingListItemName, setShoppingListItemName] = useState('');
   const [daysLeftForNextPurchase, setDaysLeftForNextPurchase] = useState(7);
+  const [shoppingListItemNameExists, setShoppingListItemNameExists] = useState(
+    false,
+  );
 
   const shoppingListItemNameHandler = (event) => {
     setShoppingListItemName(event.target.value);
+    setShoppingListItemNameExists(false);
```

Comment:
I was wondering if this line is any different from the above initialization. Since the initial state is not changing yet.

Reply:
Essentially `shoppingListItemNameExists` is supposed to be set to `true` after a user attempts to submit an item which already exists in the shopping list so that the error message is displayed on the screen declaratively. If a user starts to edit the item name, the error message has to be removed by setting the value of `shoppingListItemNameExists` back to `false`. This at the moment is done every time the user types in the input.  I am not sure whether there are performance implications if we leave it like that  but to make the intention clear I think we should change it to something like:
```js
if(shoppingListItemNameExists === true){
    setShoppingListItemNameExists(false);
}
```
This makes it clear why it is there in the first place. I don't know what you think @rb50 
Thanks for the observation @Amaka202 .

Reply:
I have implemented this in https://github.com/the-collab-lab/tcl-19-smart-shopping-list/pull/21/commits/bed277058197f45fcc71f8d536527a3c7b61bc30.

Reply:
Ok I get it. great!